service/db_test.go

```diff
@@ -0,0 +1,38 @@
+package service_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/spacemeshos/poet/service"
+)
+
+func TestInsertAndGetProof(t *testing.T) {
+	tempdir := t.TempDir()
+	proofs := make(chan service.ProofMessage)
+	db, err := service.NewProofsDatabase(tempdir, proofs)
+	require.NoError(t, err)
+
+	var eg errgroup.Group
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Just in case a test exits early because of a failed require:

```golang
defer cancel()
```

so any running go-routines get the signal to stop 🙂